src/Microsoft.DocAsCode.Build.SchemaDriven/Processors/XrefResolverInterpreter.cs

```diff
@@ -0,0 +1,118 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Build.SchemaDriven.Processors
+{
+    using System;
+
+    using Newtonsoft.Json.Schema;
+
+    using Microsoft.DocAsCode.Plugins;
+    using System.Linq;
+    using Microsoft.DocAsCode.Common;
+
+    public class XrefResolverInterpreter : IInterpreter
+    {
+        /// <summary>
+        /// Report xrefSpec when
+        /// 1. ContentType = uid is defined => internal xref spec
+        /// Or 2. XrefResolver is defined => external xref spec
+        /// </summary>
+        /// <param name="schema"></param>
+        /// <returns></returns>
+        public bool CanInterpret(BaseSchema schema)
+        {
+            if (schema.Type != JSchemaType.Object)
+            {
+                return false;
+            }
+
+            if (!string.IsNullOrEmpty(schema.XrefResolver))
+            {
+                return schema.XrefResolver.StartsWith("uid");
```

Comment:
>return schema.XrefResolver.StartsWith("uid"); [](start = 16, length = 45)

this will return true ignoring the following 2 checks?

Reply:
yes, either this is true or inner is ContentType uid

---
In reply to: [135673472](https://github.com/dotnet/docfx/pull/2008#discussion_r135673472) [](ancestors = 135673472)